lwgenerate/gcp/gcp.go

```diff
@@ -0,0 +1,505 @@
+		}
+	}
+
+	return nil
+}
+
+type GcpTerraformModifier func(c *GenerateGcpTfConfigurationArgs)
+
+// NewTerraform returns an instance of the GenerateGcpTfConfigurationArgs struct with the provided enabled
+// settings (config/audit log).
+//
+// Note: Additional configuration details may be set using modifiers of the GcpTerraformModifier type
+//
+// Basic usage: Initialize a new GcpTerraformModifier struct, with GCP service account credentials. Then use generate to
+//              create a string output of the required HCL.
+//
+//   hcl, err := gcp.NewTerraform(true, true,
+//     gcp.WithGcpServiceAccountCredentials("/path/to/sa/credentials.json")).Generate()
+//
+func NewTerraform(enableConfig bool, enableAuditLog bool, mods ...GcpTerraformModifier) *GenerateGcpTfConfigurationArgs {
+	config := &GenerateGcpTfConfigurationArgs{AuditLog: enableAuditLog, Config: enableConfig}
+	for _, m := range mods {
+		m(config)
+	}
+	return config
+}
+
+// WithGcpServiceAccountCredentials Set the path for the GCP Service Account to be utilized by the GCP provider
+func WithGcpServiceAccountCredentials(path string) GcpTerraformModifier {
+	return func(c *GenerateGcpTfConfigurationArgs) {
+		c.ServiceAccountCredentials = path
+	}
+}
+
+// WithLaceworkProfile Set the Lacework Profile to utilize when integrating
+func WithLaceworkProfile(name string) GcpTerraformModifier {
+	return func(c *GenerateGcpTfConfigurationArgs) {
+		c.LaceworkProfile = name
+	}
+}
+
+// WithOrganizationId Set the Lacework organization ID to integrate with for an organization integration
+func WithOrganizationId(id string) GcpTerraformModifier {
+	return func(c *GenerateGcpTfConfigurationArgs) {
+		c.GcpOrganizationId = id
+	}
+}
+
+// WithProjectId Set the Lacework project ID that new resources should be created in
+// (required for both project & org integration)
+func WithProjectId(id string) GcpTerraformModifier {
+	return func(c *GenerateGcpTfConfigurationArgs) {
+		c.GcpProjectId = id
+	}
+}
+
+// UseExistingServiceAccount Set an existing Service Account to be used by the Lacework Integration
+func UseExistingServiceAccount(serviceAccountDetails *ExistingServiceAccountDetails) GcpTerraformModifier {
+	return func(c *GenerateGcpTfConfigurationArgs) {
+		c.ExistingServiceAccount = serviceAccountDetails
+	}
+}
+
+// UseConfigIntegrationName Set the Config Integration name to be displayed on the Lacework UI
+func UseConfigIntegrationName(name string) GcpTerraformModifier {
+	return func(c *GenerateGcpTfConfigurationArgs) {
+		c.ConfigIntegrationName = name
+	}
+}
+
+// AuditLogLabels set labels to be applied to ALL newly created AuditLog resources
+func AuditLogLabels(labels map[string]string) GcpTerraformModifier {
+	return func(c *GenerateGcpTfConfigurationArgs) {
+		c.AuditLogLabels = labels
+	}
+}
+
+// BucketLabels set labels to be applied to the newly created AuditLog Bucket
+func BucketLabels(labels map[string]string) GcpTerraformModifier {
```

Comment:
nit: I'm wondering if the `GcpTerraformModifier` functions could have a consistent naming convention?  We start by using `With` then move to `Use` and the last set of functions don't use either

Reply:
I see the same inconsistency exists for AWS: https://github.com/lacework/go-sdk/blob/main/lwgenerate/aws/aws.go#L170

Reply:
Indeed this inconsistency exists elswhere. Will try to make a move towards standardising the naming in this class.

Reply:
My personal preference is to drop Use and With from the functions and just match the var names in the `GenerateGcpTfConfigurationArgs`

Reply:
Honestly, I could go either way; as long as we do it the same all good with me.  We can go and update the other parts of the project.

Reply:
I'd lean towards the 'With' prefix, It makes it clear these funcs are used as part of the creational pattern to construct a GenerateGcpTfConfigurationArgs

Reply:
The use of `With[XYZ]()` is what we have across this project, I think we should stick with it. 

I do agree there are inconsistencies https://github.com/lacework/go-sdk/blob/main/cli/cmd/generate.go#L83-L90 but you can see how easy it would be to know that modifiers start with `With...()` and when you start chaining them, it just makes perfect sense. 

Still, if this package follows a different pattern, I am ok with it as well. 🙌🏽

Reply:
I may as well keep us consistent with the patterns already in place. Will update to use `With...()`